domain/operation/state/prune.go

```diff
@@ -0,0 +1,97 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package state
+
+import (
+	"context"
+	"time"
+
+	"github.com/canonical/sqlair"
+	"github.com/juju/collections/transform"
+
+	"github.com/juju/juju/internal/errors"
+)
+
+// PruneOperations deletes operations older than maxAge and larger than maxSizeMB.
+func (st *State) PruneOperations(ctx context.Context, maxAge time.Duration, maxSizeMB int) error {
+
+	// Prune by age, completed only
+	err := st.pruneCompletedOperationsOlderThan(ctx, maxAge)
+	if err != nil {
+		return errors.Errorf("pruning completed operation older than %s: %w", maxAge, err)
+	}
+
+	// Prune by size, completed only
+	// TODO(gfouillet): followup PR
+
+	// Prune by size, any action
+	// TODO(gfouillet): followup PR
```

Comment:
question: please expand on what this todo covers. 

Only completed operations are pruned.